pkg/component/runtime/manager.go

```diff
@@ -705,6 +705,12 @@ func (m *Manager) update(model component.Model, teardown bool) error {
 		var stoppedWg sync.WaitGroup
 		stoppedWg.Add(len(stop))
 		for _, existing := range stop {
+			if existing.getLatest().State == client.UnitStateStopped {
```

Comment:
I think instead of making this check here this could be changed to use the atomic that the `componentRuntimeState` is already setting.

https://github.com/elastic/elastic-agent/blob/d316ee0353e08eba4e45cf584efabb1833c358fe/pkg/component/runtime/runtime.go#L190

I think change line 190 in `runtime.go` to check if already set and do nothing would result in the same behavior. Removing the need to hold the lock that is being held when `getLatest()` is being called.